File: 0009-palindrome-number/0009-palindrome-number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    if x < 0 
        return false
    end
    result = 0
    original = x
    value = 0
    while x > 0 
        value = x % 10
        x = x / 10
        result = result * 10 + value
    end
    if result == original
        return true
    end
    return false
end

File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if ( x < 0 ) return false;
    let result = 0;
    let org = x;
    let lastDigit = 0;
    while ( x > 0 ) {
        lastDigit = x % 10;
        x = parseInt(x / 10);
        result = 10 * result + lastDigit;
    }
    if ( result === org ) {
        return true;
    }
    return false;
};